sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp

```diff
@@ -7,6 +7,10 @@
 // ===--------------------------------------------------------------------=== //
 
 #pragma once
+
+#include <CL/__spirv/spirv_types.hpp> // __spv namespace
+#include <optional>                   // std::optional
```

Comment:
what is the reason behind this change?

Reply:
This matrix code was getting std::optional and spv namespace by virtue of an inclusion to bfloat16.hpp which in turn included the math builtins which included those things. 
But now bfloat16.hpp is a dependency OF the math builtins and itself no longer includes them (otherwise cyclic dependency). So this matrix file now needs to include the resources it uses directly.